server/incore-common/src/main/java/edu/illinois/ncsa/incore/common/utils/JsonUtils.java

```diff
@@ -49,4 +61,149 @@ public static String extractValueFromJsonString(String inId, String inJson) {
             return outList;
         }
     }
+
+    public static JSONObject createUserFinalQuotaJson(String username, IUserFinalQuotaRepository finalQuotaRepository) throws ParseException{
+        UserFinalQuota quota = finalQuotaRepository.getQuotaByUsername(username);   // get default allocation
+        JSONObject outJson = new JSONObject();
+        if (quota != null) {
+            // get user's limit
+            UserUsages limit = new UserUsages();
+            limit = quota.getApplicationLimits();
+
+            // check if there is the correct limit values is there, otherwise give default values
+            if (limit.getDatasets() == 0) {
```

Comment:
why is this check needed? what if we really want to set a limit to 0?

Reply:
this is because mongo UserFinalQuota view results are null in most case. If it is null, it shows it as zero. If we update the script and code, then it will be the real values, then we can remove this part. However, currently, we need to set this otherwise, every one's allocation will be zero unitl the UserFinalQuota view gets completely funcational

Reply:
I just saw your other comment with the screen shot and it is related to that one. If there is a problem in that comment and if it get fixed, then this can be removed.

Reply:
This needs to be there for a while maybe. Even I changed the problem in the one that you made a screenshot below, the problem of having null values in UserFinalQuota still exists in most records and this will be there until we develop the script if I understand the original plan correctly.